server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -271,6 +275,35 @@ public void onFailure(Exception e) {
         }
     }
 
+    static void prohibitAppendOnlyWritesInBackingIndices(DocWriteRequest<?> writeRequest, Metadata metadata) {
```

Comment:
nit: I think

```suggestion
    static void prohibitAppendWritesInBackingIndices(DocWriteRequest<?> writeRequest, Metadata metadata) {
```

is more precise in that we try to prohibit any direct append into background indices?